### PR TITLE
Avoid mailing users that have explicitly unwatched an issue

### LINKED ISSUE #8879: I'm still receiving email notifications for issue comments while not watching repo and not being subscribed to issue
I saw this happening on gitea.com, but cant remember the exact issue it was occurring for.
@lunny might recall

### PR DESCRIPTION
**Pull request by 6543**
close #8879

### CODE REVIEW

**Comment by 6543** [comment]
bug exist in 1.11.x !

File: services/mailer/mail_issue.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ func mailIssueCommentToParticipants(ctx *mailCommentContext, mentions []int64) e
 		return fmt.Errorf("LoadPullRequest(): %v", err)
 	}
 
-	// Enough room to avoid reallocations
+	// Enough room to avoid reallocation's
```
Comment (guillep2k) [inline services/mailer/mail_issue.go]:
```suggestion
	// Enough room to avoid reallocations
```
reallocations is correct

**Review by guillep2k — APPROVED** [review]
Just a small nit. Otherwise LGTM.

**Comment by guillep2k** [comment]
I'm sorry for having introduced this bug. 😔

**Comment by 6543** [comment]
@guillep2k I think this bug was there before you changed some code ...

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
Please send backport

**Comment by 6543** [comment]
backport: #10500

**Pull request merged**